app/models/work.rb

```diff
@@ -0,0 +1,30 @@
+class Work < ApplicationRecord
+  validates :category, presence: true
+  validates :title, presence: true, uniqueness: true
+  has_many :votes, dependent: :destroy
+
+  def self.top_ten(type)
+    array_by_type = self.where(category: type)
+
+    if array_by_type.length > 1
+      array_by_votes = array_by_type.sort_by { |work| -work.votes.count }
+
+      if array_by_votes.length < 10
+        return array_by_votes
+      else
+        return array_by_votes[0..9]
+      end
+    else
+      return array_by_type
```

Comment:
Hm... `array_by_type` (since it comes from a `where` call) will always be an array... even an empty array. If that's the case, do we need the `if ... else` here? Consider the following refactor of the method:

```ruby
def self.top_ten(type)
    array_by_type = self.where(category: type)
    array_by_votes = array_by_type.sort_by { |work| -work.votes.count }
    return array_by_votes[0..9]
end
```

The tests still pass! When you are ready, take some time to think about why this still works and is effectively the same thing as your current code